pkg/trace/writer/trace.go

```diff
@@ -91,7 +91,12 @@ func NewTraceWriter(cfg *config.AgentConfig, in <-chan *SampledSpans) *TraceWrit
 	qsize := cfg.TraceWriter.QueueSize
 	if qsize == 0 {
 		// default to 50% of maximum memory.
-		qsize = int(math.Max(1, cfg.MaxMemory/2/float64(maxPayloadSize)))
+		maxmem := cfg.MaxMemory / 2
```

Comment:
Nit: Would it make sense to have `cfg.MaxTraceWriteMemory` and `cfg.MaxStatsWriterMemory` encoding this factor of 2 vs factor of 4 logic all in the same place instead of spreading it around?